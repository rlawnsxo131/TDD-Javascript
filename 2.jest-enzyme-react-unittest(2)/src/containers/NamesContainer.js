import React, { Component, Fragment } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import NamesForm from '../components/NameForm';
import NameList from '../components/NameList';
import * as namesActions from '../store/modules/names';

class NamesContainer extends Component {
  onChange = (e) => {
    const { NamesActions } = this.props;
    NamesActions.changeInput(e.target.value); 
  }
  onSubmit = (e) => {
    const { NamesActions, input } = this.props;
    NamesActions.insert(input);
    NamesActions.changeInput('');
    e.preventDefault();
  }
  
  render() {
    const { onChange, onSubmit } = this;
    const { names, input } = this.props;
    return (
      <Fragment>
        <NamesForm value={input} onChange={onChange} onSubmit={onSubmit} />
        <NameList names={names} />
      </Fragment>
    );
  }
}

export default connect(
  (state) => ({
    names: state.names.names,
    input: state.names.input
  }),
  (dispatch) => ({
    NamesActions: bindActionCreators(namesActions, dispatch)
  })
)(NamesContainer);
