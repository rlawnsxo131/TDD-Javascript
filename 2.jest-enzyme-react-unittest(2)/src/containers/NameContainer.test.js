import React from 'react';
import {Provider} from 'react-redux';
import { mount } from 'enzyme';
import NamesContainer from './NamesContainer';
import configureStore from '../store/configureStore';

describe('NamesContainer', () => {
  let component = null;

  // 이번엔 실제 store 로 테스팅을 한다.
  let store = configureStore();

  it('renders properly', () => {
    component = mount(
      <Provider store={store}>
        <NamesContainer />
      </Provider>
    );
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('dispatches CHANGE_INPUT action', () => {
    const mockedEvent = {
      target: {
        value: 'world'
      }
    };
    component.find('input').simulate('change', mockedEvent);
    expect(store.getState().names.input).toBe('world');
  });

  it('dispatches INSERT action', () => {
    component.find('form').simulate('submit');
    expect(store.getState().names.names).toEqual(['world']);
  });
});
