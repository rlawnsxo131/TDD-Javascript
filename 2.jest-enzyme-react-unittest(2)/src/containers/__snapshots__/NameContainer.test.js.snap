// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NamesContainer matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(NamesContainer)>
    <NamesContainer
      NamesActions={
        Object {
          "changeInput": [Function],
          "default": [Function],
          "insert": [Function],
        }
      }
      input=""
      names={Array []}
    >
      <NameForm
        onChange={[Function]}
        onSubmit={[Function]}
        value=""
      >
        <form
          onSubmit={[Function]}
        >
          <label>
            이름
          </label>
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            등록
          </button>
        </form>
      </NameForm>
      <NameList
        names={Array []}
      >
        <ul />
      </NameList>
    </NamesContainer>
  </Connect(NamesContainer)>
</Provider>
`;
